community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactions.java

```diff
@@ -219,13 +226,15 @@ public boolean haveClosingTransaction()
     @Override
     public void start() throws Throwable
     {
+        stopped = false;
         unblockNewTransactions();
     }
 
     @Override
     public void stop() throws Throwable
     {
         blockNewTransactions();
+        stopped = true;
```

Comment:
Why is this volatile necessary when we already have the `newTransactionsLock`? This lock, by the way, could be changed to a `StampedLock` – it's not reentrant, so there would not be a need for the `assertCurrentThreadIsNotBlockingNewTransactions()` check.